hedera-node/hedera-smart-contract-service-impl/src/main/java/com/hedera/node/app/service/contract/impl/exec/utils/FrameBuilder.java

```diff
@@ -159,18 +157,31 @@ private MessageFrame finishedAsCall(
             @NonNull final ContractsConfig config) {
         Code code = CodeV0.EMPTY_CODE;
         final var contractId = transaction.contractIdOrThrow();
+        final var contractMustBePresent = contractMustBePresent(config, featureFlags, contractId);
 
-        if (canLoadCodeFromAccount(transaction, worldUpdater, contractId, config, featureFlags)) {
-            final var account = worldUpdater.getHederaAccount(to);
-            if (account == null && contractMustBePresent(config, featureFlags, contractId)) {
-                validateTrue(transaction.permitsMissingContract(), INVALID_ETHEREUM_TRANSACTION);
+        // If the contract has been deleted, then always use empty byte code
```

Comment:
Comment at L162 doesn't agree with comment at L180/182.  Or maybe it's just in the wrong place.

Reply:
In fact I would rewrite with quick exit on deleted contract first, something like

```java
if (contractDeleted(worldUpdater, contractId)) {
    validateTrue(....);
   return buildMessageCall(to, transaction, CodeV0.EMPTY_CODE);
}
final var account ...
if (account == null) {
   if (contractMustBePresent) validateTrue(...);
   return buildMessageCall(to, transaction, CodeV0.EMPTY_CODE);
}
// Now we've got a valid contract, continue:
...
```

This unnests everything and leaves the remaining code as the happy path which will be easy to read.

Reply:
Nice suggestion!